lib/davinci_dtr_test_kit/client_groups/full_ehr/dtr_full_ehr_questionnaire_response_correctness_test.rb

```diff
@@ -1,34 +1,36 @@
 require_relative '../../dtr_questionnaire_response_validation'
+require_relative '../../cql_test'
 
 module DaVinciDTRTestKit
   class DTRFullEHRQuestionnaireResponseCorrectnessTest < Inferno::Test
     include DTRQuestionnaireResponseValidation
+    include DaVinciDTRTestKit::CQLTest
 
     id :dtr_full_ehr_questionnaire_response_correctness
     title 'QuestionnaireResponse is correct for the workflow'
     description %(
-      The QuestionnaireResponse aligns with the following expectations for the workflow. This includes checks for
-      the presence of the following answers and their appropriate origin source extensions:
-
-      - `PBD.1` (Last Name) and `LOC.1` (Location): `auto`
-      - PBD.2 (First Name): `override`
-      - `3` (all nested dinner questions): `manual`
+      This test verifies that all the QuestionnaireResponse.item have the appropriate
+      origin source extensions and that all required questions have been answered.
     )
 
     run do
-      skip_if questionnaire_response.nil?, 'Completed QuestionnaireResponse input was blank'
+      skip_if questionnaire_response.blank?, 'Completed QuestionnaireResponse input was blank'
       check_is_questionnaire_response(questionnaire_response)
 
       qr = FHIR.from_contents(questionnaire_response)
-      questionnaire = if config.options[:adaptive]
-                        qr.contained.find do |res|
-                          res.resourceType == 'Questionnaire'
-                        end
-                      else
+      questionnaire = if scratch[:static_questionnaire_bundles].nil?
                         Fixtures.questionnaire_for_test(id)
+                      else
+                        questionnaires = extract_questionnaires_from_bundles(scratch[:static_questionnaire_bundles])
+                        questionnaires.find { |q| qr.questionnaire.end_with?(q.id) }
                       end
-      # questionnaire = Fixtures.find_questionnaire('DinnerOrderStatic')
-      check_origin_sources(questionnaire.item, qr.item, expected_overrides: ['PBD.2'])
+
+      skip_if questionnaire.blank?, "Couldn't find Questionnaire #{qr.questionnaire} to check the QuestionnaireResponse"
+
+      expected_overrides = scratch[:static_questionnaire_bundles].nil? ? [] : ['PBD.2']
```

Comment:
Shouldn't we no longer be checking the 'PBD.2' override?

Reply:
Actually, I reversed the logic here. What I intended was to check the 'PBD.2' override if a custom response is not provided and skip checking it if a custom response is supplied (`scratch[:static_questionnaire_bundles].nil? ? ['PBD.2'] : []`). We can also choose not to check it at all if that makes more sense.

Reply:
Given this change, I think we should remove the override check

<img width="829" alt="image" src="https://github.com/user-attachments/assets/5bb635b5-78eb-4001-b0b1-22d80b1c736c" />